plutus-metatheory/src/Algorithmic.lagda

```diff
@@ -272,11 +200,11 @@ data Arg : Set where
 
 Arity = List Arg
 
-ctx2bwdarity : ∀{Φ}(Γ : Ctx Φ) → Bwd Arg
-ctx2bwdarity ∅        = []
-ctx2bwdarity (Γ ,⋆ J) = ctx2bwdarity Γ :< Type
-ctx2bwdarity (Γ , A)  = ctx2bwdarity Γ :< Term
-
 arity : Builtin → Arity
-arity b = ctx2bwdarity (proj₁ (proj₂ (sig b))) <>> []
+arity b with signature b 
+... | sig n ar _ = replicate n Type ++ replicate (length ar) Term
+
 \end{code}
+  
+Arity should be rewritten to be two numbers (`n` and `length ar` above). 
```

Comment:
Perhaps mention that that current representation supports interleaving but when this is removed only numbers will be needed.

Reply:
Done.